source/develop/developer-guide/vdsm/hooks.html.md

```diff
@@ -207,33 +180,45 @@ This xml can be changed in order to alter the VM settings in ways ovirt-engine d
 
 Currently, the hooking module only exists in python. Here is an example of a simple vdsm hook that blocks a VM's live migration: 50_cant_migrate.py placed in before_VM_migrate_source. This hook looks for a custom property calles "cantmigrate" and if it is found, will block the VM from migrating. This can be useful for VMs that rely on a specific resource found only locally on a specific host (like a passed through USB license dongle or a TBU)
 
-      #!/usr/bin/python
-      import os
-      import sys 
-      if os.environ.has_key('cantmigrate'):
-         sys.stderr.write('cantmigrate: before_vm_migrate_source: cannot migrate this VM\n')
-         sys.exit(2)
+```python
+  #!/usr/bin/python
+  import os
+  import sys 
+  if os.environ.has_key('cantmigrate'):
+     sys.stderr.write('cantmigrate: before_vm_migrate_source: cannot migrate this VM\n')
+     sys.exit(2)
+```
 
 This script does not use the hooking module, because it does not need to actually act upon the VM's specific XML.
 
 And this is an example that uses the Hooking module to read a custom property that defines CPU pinning for a VM, and alters the VM's XML accordingly:
 
-      #!/usr/bin/python 
-      import os
-      import sys
-      import hooking
-      import traceback
-      **\1**
-      if os.environ.has_key('pincpu'):
-          try:
-`       domxml = hooking.read_domxml()   `**`#here` `we` `read` `the` `VM` `XML` `into` `the` `domxml` `variable`**
-`       vcpu = domxml.getElementsByTagName('vcpu')[0] `**`#find` `and` `read` `the` `CPU` `definition` `in` `the` `VM` `XML`**
-             if not vcpu.hasAttribute('cpuset'):
-`           sys.stderr.write('pincpu: pinning cpu to: %s\n' % os.environ['pincpu'])  `**`#sys.stderr.write` `is` `caught` `by` `vdsm` `and` `logged` `into` `vdsm.log` `for` `debugging`**
-`           vcpu.setAttribute('cpuset', os.environ['pincpu'])  `**`#change` `an` `attribute` `here`**
-`           hooking.write_domxml(domxml)                       `**`#and` `write` `to` `the` `altered` `domxml`**
-             else:
-                 sys.stderr.write('pincpu: cpuset attribute is present in vcpu, doing nothing\n')
-         except:
-             sys.stderr.write('pincpu: [unexpected error]: %s\n' % traceback.format_exc())
-             sys.exit(2)
+```python
+  #!/usr/bin/python 
+  import os
+  import sys
+  import hooking
+  import traceback
+      
+  pincpu usages
+  =============
```

Comment:
This should be an executable python script people can copy-paste from the site. Please comment out the "pincpu usages" 'header'